java/java-tests/testData/inspection/dataFlow/fixture/ConsumedStream.java

```diff
@@ -0,0 +1,191 @@
+import java.util.Iterator;
+import java.util.Random;
+import java.util.stream.Stream;
+
+public class ConsumedStream {
+  public static void test1() {
+    Stream<String> stream = Stream.of("x");
+    stream.forEach(System.out::println);
+    <warning descr="Stream has already been linked or consumed">stream</warning>.forEach(System.out::println);
+  }
+
+  public static void test2() {
+    Stream<String> stream = Stream.of("x");
+    Stream<String> stream2 = stream.filter(String::isEmpty);
+    Stream<String> stream3 = <warning descr="Stream has already been linked or consumed">stream</warning>.filter(x -> !x.isEmpty());
+  }
+
+  public static void test3() {
+    Stream<String> stream = Stream.of("x");
+    for (int i = 0; i < 10; i++) {
+      <warning descr="Stream has already been linked or consumed">stream</warning>.forEach(System.out::println);
```

Comment:
This message sounds strange, as on the first iteration it's not true. I think we should separate cases when we have 100% problem and we might have a problem. There should be different messages and "might" problems should be suppressed if REPORT_UNSOUND_WARNINGS is false. You can see how it's done e.g. in `reportFailingCasts`.

Reply:
Fixed. Thank you!